snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/tracker/SessionState.java

```diff
@@ -13,41 +13,46 @@ public class SessionState implements State {
 
     @NonNull
     private final String firstEventId;
-    @Nullable
-    private final String previousSessionId;
+    @NonNull
+    private final String firstEventTimestamp;
     @NonNull
     private final String sessionId;
+    @Nullable
+    private final String previousSessionId;
     private final int sessionIndex;
     @NonNull
-    private final String storage;
-    @NonNull
     private final String userId;
+    @NonNull
+    private final String storage;
 
     @NonNull
-    private Map<String, Object> sessionContext;
+    private final Map<String, Object> sessionContext;
```

Comment:
👍 Nice add! - btw, maybe you are already aware of this but, just in case... making it final will not make the map immutable. It will make immutable only the reference to the map.

Reply:
Yeah that makes sense. IntelliJ was hassling me to make it final.